pwndbg/lib/regs.py

```diff
@@ -75,10 +75,12 @@ def __init__(
         self.retval = retval
 
         # In 'common', we don't want to lose the ordering of:
-        self.common = []
+        self.common_no_flag = []
         for reg in gpr + (frame, stack, pc) + tuple(flags):
-            if reg and reg not in self.common:
-                self.common.append(reg)
+            if reg and reg not in self.common_no_flag:
+                self.common_no_flag.append(reg)
+
+        self.common = self.common_no_flag + list(flags)
```

Comment:
This was the fix that involved the least code changes I could think of - I'm all ears for any other way of separating all the relevant registers (while removing duplicates) from the flag register.